syntax = "proto2";

message KeyReference {
	required bytes wrappedKey = 1;
	optional int32 rfu = 2;
}

message SecretData {
	required bytes encryptedData = 1;
	required KeyReference keyReference = 2;
	required string tamperCheck = 3;	
}

message EncryptedMetadata {
	required bytes encryptedMetadata = 1;
	required bytes encryptedMetadataKey = 2;
	required string tamperCheck = 3; 
} 

message ItemV8 {
	required SecretData secretData = 1;
	required EncryptedMetadata encryptedMetadata = 2;

    enum Keyclass {
        KEYCLASS_AK = 6;
        KEYCLASS_CK = 7;
        KEYCLASS_DK = 8;
        KEYCLASS_AKU = 9;
        KEYCLASS_CKU = 10;
        KEYCLASS_DKU = 11;
        KEYCLASS_AKPU = 12;
    }
    required Keyclass keyclass = 3 [default = KEYCLASS_AKPU];
}

